src/test/java/jenkins/plugins/office365connector/ValidateUrlTest.java

```diff
@@ -0,0 +1,38 @@
+package jenkins.plugins.office365connector;
+
+import java.util.Arrays;
+import java.util.Collection;
+
+import jenkins.plugins.office365connector.util.FormUtils;
+import org.junit.Test;
+import org.junit.runner.RunWith;
+import org.junit.runners.Parameterized;
+
+import static org.assertj.core.api.Assertions.assertThat;
+
+@RunWith(Parameterized.class)
+public class ValidateUrlTest {
+
+    @Parameterized.Parameter()
+    public String input;
+    @Parameterized.Parameter(value = 1)
+    public boolean expected;
+
+    @Parameterized.Parameters
+    public static Collection<Object[]> data() {
+        return Arrays.asList(new Object[][]{
+                {"demo", false},
+                {"hpp://demo", false},
+                {"http://demo", false},
+                /* Still need to figure a way out with this*/// { "$$$$$demo", false },
+                {"$demo", true},
+                {"https://demo.com", true},
+        });
+    }
+
+    @Test
+    public void shouldValidateUrlOrVariableReference() throws Exception {
```

Comment:
Looks like you copied&paste the code and ``throws Exception`` in signature is present even not needed

Same for other methods